pkg/metrics/metrics.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2018 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metrics
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+
+	"github.com/operator-framework/operator-sdk/pkg/k8sutil"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	v1 "k8s.io/api/core/v1"
+)
+
+// ExposeMetricsPort generate a Kubernetes Service to expose metrics port
+func ExposeMetricsPort() (*v1.Service, error) {
+	reg := NewOperatoSDKPrometheusRegistery()
+	mux := http.NewServeMux()
+	HandlerFuncs(mux, reg)
+	go log.Println(http.ListenAndServe(":"+strconv.Itoa(OperatorSDKPrometheusMetricsPort), mux))
+
+	service, err := k8sutil.InitOperatorService(int32(OperatorSDKPrometheusMetricsPort), OperatorSDKPrometheusMetricsPortName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize service object for operator metrics: %v", err)
+	}
+	return service, nil
+}
+
+// NewOperatoSDKPrometheusRegistery returns a newly created prometheus registry.
+// It also registers go collector and process collector metrics.
+func NewOperatoSDKPrometheusRegistery() *prometheus.Registry {
```

Comment:
I am also wondering why we need to `OperatorSDK` registry prefix here, will that not be apparent via the import? basically same nit as above.

Reply:
Done, makes perfect sense, not a nit at all!